### ISSUE TITLE
TypeError: train_step() got an unexpected keyword argument 'running_status'

### ISSUE THREAD

**Issue by Lasia205565** [open-mmlab/mmgeneration#36]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmgeneration.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
3. The bug has not been fixed in the latest version.

**Describe the bug**

There is something wrong with the DDP. I'm trying to train a pix2pix model with mmgeneration and it reports error. However, when I train a pix2pix model with mmediting, it does not has this issue. I believe that this issue is caused by the dynamic iteration.

**Reproduction**

1. What command or script did you run?
bash tools/dist_train.sh configs/pix2pix/pix2pix_vanilla_unet_bn_1x1_80k_facades.py 1

```none
A placeholder for the command.
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
No.
3. What dataset did you use?
Facades

**Environment**
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
sys.platform: linux
Python: 3.7.1 (default, Dec 14 2018, 19:28:38) [GCC 7.3.0]
CUDA available: True
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.2, V10.2.89
GPU 0,1,2,3,4,5,6,7: GeForce RTX 2080 Ti
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
PyTorch: 1.5.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.1 Product Build 20180928 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.6.0
OpenCV: 4.1.1
MMCV: 1.3.5
MMGen: 0.2.0+
MMCV Compiler: GCC 5.4
MMCV CUDA Compiler: 10.2
1. Please run `python mmgen/utils/collect_env.py` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - with pip. I have tried torch==1.5.0 and 1.8.1. Both does not work.
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error trackback here.
Python: 3.7.1 (default, Dec 14 2018, 19:28:38) [GCC 7.3.0]
CUDA available: True
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.2, V10.2.89
GPU 0: GeForce RTX 2080 Ti
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
PyTorch: 1.5.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.1 Product Build 20180928 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

TorchVision: 0.6.0
OpenCV: 4.1.1
MMCV: 1.3.5
MMGen: 0.2.0+
MMCV Compiler: GCC 5.4
MMCV CUDA Compiler: 10.2
------------------------------------------------------------

2021-05-31 15:45:07,821 - mmgen - INFO - Distributed training: True
2021-05-31 15:45:08,123 - mmgen - INFO - Config:
model = dict(
    type='Pix2Pix',
    generator=dict(
        type='UnetGenerator',
        in_channels=3,
        out_channels=3,
        num_down=8,
        base_channels=64,
        norm_cfg=dict(type='BN'),
        use_dropout=True,
        init_cfg=dict(type='normal', gain=0.02)),
    discriminator=dict(
        type='PatchDiscriminator',
        in_channels=6,
        base_channels=64,
        num_conv=3,
        norm_cfg=dict(type='BN'),
        init_cfg=dict(type='normal', gain=0.02)),
    gan_loss=dict(
        type='GANLoss',
        gan_type='vanilla',
        real_label_val=1.0,
        fake_label_val=0.0,
        loss_weight=1.0),
    pixel_loss=dict(type='L1Loss', loss_weight=100.0, reduction='mean'))
train_cfg = dict(direction='b2a')
test_cfg = dict(direction='b2a', show_input=True)
train_dataset_type = 'PairedImageDataset'
val_dataset_type = 'PairedImageDataset'
img_norm_cfg = dict(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])
train_pipeline = [
    dict(
        type='LoadPairedImageFromFile',
        io_backend='disk',
        key='pair',
        flag='color'),
    dict(
        type='Resize',
        keys=['img_a', 'img_b'],
        scale=(286, 286),
        interpolation='bicubic'),
    dict(type='FixedCrop', keys=['img_a', 'img_b'], crop_size=(256, 256)),
    dict(type='Flip', keys=['img_a', 'img_b'], direction='horizontal'),
    dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
    dict(
        type='Normalize',
        keys=['img_a', 'img_b'],
        to_rgb=False,
        mean=[0.5, 0.5, 0.5],
        std=[0.5, 0.5, 0.5]),
    dict(type='ImageToTensor', keys=['img_a', 'img_b']),
    dict(
        type='Collect',
        keys=['img_a', 'img_b'],
        meta_keys=['img_a_path', 'img_b_path'])
]
test_pipeline = [
    dict(
        type='LoadPairedImageFromFile',
        io_backend='disk',
        key='pair',
        flag='color'),
    dict(
        type='Resize',
        keys=['img_a', 'img_b'],
        scale=(256, 256),
        interpolation='bicubic'),
    dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
    dict(
        type='Normalize',
        keys=['img_a', 'img_b'],
        to_rgb=False,
        mean=[0.5, 0.5, 0.5],
        std=[0.5, 0.5, 0.5]),
    dict(type='ImageToTensor', keys=['img_a', 'img_b']),
    dict(
        type='Collect',
        keys=['img_a', 'img_b'],
        meta_keys=['img_a_path', 'img_b_path'])
]
data = dict(
    samples_per_gpu=1,
    workers_per_gpu=4,
    drop_last=True,
    train=dict(
        type='PairedImageDataset',
        dataroot='data/paired/facades',
        pipeline=[
            dict(
                type='LoadPairedImageFromFile',
                io_backend='disk',
                key='pair',
                flag='color'),
            dict(
                type='Resize',
                keys=['img_a', 'img_b'],
                scale=(286, 286),
                interpolation='bicubic'),
            dict(
                type='FixedCrop',
                keys=['img_a', 'img_b'],
                crop_size=(256, 256)),
            dict(type='Flip', keys=['img_a', 'img_b'], direction='horizontal'),
            dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
            dict(
                type='Normalize',
                keys=['img_a', 'img_b'],
                to_rgb=False,
                mean=[0.5, 0.5, 0.5],
                std=[0.5, 0.5, 0.5]),
            dict(type='ImageToTensor', keys=['img_a', 'img_b']),
            dict(
                type='Collect',
                keys=['img_a', 'img_b'],
                meta_keys=['img_a_path', 'img_b_path'])
        ],
        test_mode=False),
    val=dict(
        type='PairedImageDataset',
        dataroot='data/paired/facades',
        pipeline=[
            dict(
                type='LoadPairedImageFromFile',
                io_backend='disk',
                key='pair',
                flag='color'),
            dict(
                type='Resize',
                keys=['img_a', 'img_b'],
                scale=(256, 256),
                interpolation='bicubic'),
            dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
            dict(
                type='Normalize',
                keys=['img_a', 'img_b'],
                to_rgb=False,
                mean=[0.5, 0.5, 0.5],
                std=[0.5, 0.5, 0.5]),
            dict(type='ImageToTensor', keys=['img_a', 'img_b']),
            dict(
                type='Collect',
                keys=['img_a', 'img_b'],
                meta_keys=['img_a_path', 'img_b_path'])
        ],
        test_mode=True),
    test=dict(
        type='PairedImageDataset',
        dataroot='data/paired/facades',
        pipeline=[
            dict(
                type='LoadPairedImageFromFile',
                io_backend='disk',
                key='pair',
                flag='color'),
            dict(
                type='Resize',
                keys=['img_a', 'img_b'],
                scale=(256, 256),
                interpolation='bicubic'),
            dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
            dict(
                type='Normalize',
                keys=['img_a', 'img_b'],
                to_rgb=False,
                mean=[0.5, 0.5, 0.5],
                std=[0.5, 0.5, 0.5]),
            dict(type='ImageToTensor', keys=['img_a', 'img_b']),
            dict(
                type='Collect',
                keys=['img_a', 'img_b'],
                meta_keys=['img_a_path', 'img_b_path'])
        ],
        test_mode=True))
checkpoint_config = dict(interval=4000, by_epoch=False, save_optimizer=True)
log_config = dict(
    interval=100, hooks=[dict(type='TextLoggerHook', by_epoch=False)])
custom_hooks = [
    dict(
        type='VisualizationHook',
        output_dir='training_samples',
        res_name_list=['fake_b'],
        interval=100)
]
runner = dict(
    type='DynamicIterBasedRunner',
    is_dynamic_ddp=True,
    pass_training_status=True)
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
find_unused_parameters = True
cudnn_benchmark = True
dataroot = 'data/paired/facades'
optimizer = dict(
    generator=dict(type='Adam', lr=0.0002, betas=(0.5, 0.999)),
    discriminator=dict(type='Adam', lr=0.0002, betas=(0.5, 0.999)))
lr_config = None
visual_config = None
total_iters = 80000
exp_name = 'pix2pix_facades'
work_dir = './work_dirs/experiments/pix2pix_facades'
metrics = dict(
    FID=dict(type='FID', num_images=106, image_shape=(3, 256, 256)),
    IS=dict(type='IS', num_images=106, image_shape=(3, 256, 256)))
gpu_ids = range(0, 1)

2021-05-31 15:45:08,124 - mmgen - INFO - Set random seed to 2021, deterministic: False
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
2021-05-31 15:45:13,765 - mmgen - INFO - Start running, host: lthpc@lthpc, work_dir: /home/lthpc/mmgeneration-master/work_dirs/experiments/pix2pix_facades
2021-05-31 15:45:13,765 - mmgen - INFO - workflow: [('train', 1)], max: 80000 iters
Traceback (most recent call last):
  File "tools/train.py", line 163, in <module>
    main()
  File "tools/train.py", line 159, in main
    meta=meta)
  File "/home/lthpc/mmgeneration-master/mmgen/apis/train.py", line 196, in train_model
    runner.run(data_loaders, cfg.workflow, cfg.total_iters)
  File "/home/lthpc/mmgeneration-master/mmgen/core/runners/dynamic_iterbased_runner.py", line 284, in run
    iter_runner(iter_loaders[i], **kwargs)
  File "/home/lthpc/mmgeneration-master/mmgen/core/runners/dynamic_iterbased_runner.py", line 214, in train
    outputs = self.model.train_step(data_batch, self.optimizer, **kwargs)
  File "/home/lthpc/anaconda3/lib/python3.7/site-packages/mmcv/parallel/distributed.py", line 51, in train_step
    output = self.module.train_step(*inputs[0], **kwargs[0])
TypeError: train_step() got an unexpected keyword argument 'running_status'
Traceback (most recent call last):
  File "/home/lthpc/anaconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/lthpc/anaconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/lthpc/anaconda3/lib/python3.7/site-packages/torch/distributed/launch.py", line 263, in <module>
    main()
  File "/home/lthpc/anaconda3/lib/python3.7/site-packages/torch/distributed/launch.py", line 259, in main
    cmd=cmd)
subprocess.CalledProcessError: Command '['/home/lthpc/anaconda3/bin/python', '-u', 'tools/train.py', '--local_rank=0', 'configs/pix2pix/pix2pix_vanilla_unet_bn_1x1_80k_facades.py', '--launcher', 'pytorch']' returned non-zero exit status 1.
```none
A placeholder for trackback.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by nbei** [comment]
You can quickly fix this bug by adding the following lines to your config file. 

```python
runner = None
use_ddp_wrapper = True
```

In the near future, @plyfager will fix it.

**Comment by plyfager** [comment]
> Thanks for your error report and we appreciate it a lot.
> 
> **Checklist**
> 
> 1. I have searched related issues but cannot get the expected help.
> 2. I have read the [FAQ documentation](https://mmgeneration.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
> 3. The bug has not been fixed in the latest version.
> 
> **Describe the bug**
> A clear and concise description of what the bug is.
> 
> There is something wrong with the DDP. I'm trying to train a pix2pix model with mmgeneration and it reports error. However, when I train a pix2pix model with mmediting, it does not has this issue. I believe that this issue is caused by the dynamic iteration.
> 
> **Reproduction**
> 
> 1. What command or script did you run?
>    bash tools/dist_train.sh configs/pix2pix/pix2pix_vanilla_unet_bn_1x1_80k_facades.py 1
> 
> ```
> A placeholder for the command.
> ```
> 
> 1. Did you make any modifications on the code or config? Did you understand what you have modified?
>    No.
> 2. What dataset did you use?
>    Facades
> 
> **Environment**
> fatal: Not a git repository (or any parent up to mount point /home)
> Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
> sys.platform: linux
> Python: 3.7.1 (default, Dec 14 2018, 19:28:38) [GCC 7.3.0]
> CUDA available: True
> CUDA_HOME: /usr/local/cuda
> NVCC: Cuda compilation tools, release 10.2, V10.2.89
> GPU 0,1,2,3,4,5,6,7: GeForce RTX 2080 Ti
> GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
> PyTorch: 1.5.0
> PyTorch compiling details: PyTorch built with:
> 
> * GCC 7.3
> * C++ Version: 201402
> * Intel(R) Math Kernel Library Version 2019.0.1 Product Build 20180928 for Intel(R) 64 architecture applications
> * Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
> * OpenMP 201511 (a.k.a. OpenMP 4.5)
> * NNPACK is enabled
> * CPU capability usage: AVX2
> * CUDA Runtime 10.2
> * NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
> * CuDNN 7.6.5
> * Magma 2.5.2
> * Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,
> 
> TorchVision: 0.6.0
> OpenCV: 4.1.1
> MMCV: 1.3.5
> MMGen: 0.2.0+
> MMCV Compiler: GCC 5.4
> MMCV CUDA Compiler: 10.2
> 
> 1. Please run `python mmgen/utils/collect_env.py` to collect necessary environment information and paste it here.
> 2. You may add addition that may be helpful for locating the problem, such as
>    
>    * How you installed PyTorch [e.g., pip, conda, source]
>    * with pip. I have tried torch==1.5.0 and 1.8.1. Both does not work.
>    * Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)
> 
> **Error traceback**
> If applicable, paste the error trackback here.
> Python: 3.7.1 (default, Dec 14 2018, 19:28:38) [GCC 7.3.0]
> CUDA available: True
> CUDA_HOME: /usr/local/cuda
> NVCC: Cuda compilation tools, release 10.2, V10.2.89
> GPU 0: GeForce RTX 2080 Ti
> GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
> PyTorch: 1.5.0
> PyTorch compiling details: PyTorch built with:
> 
> * GCC 7.3
> * C++ Version: 201402
> * Intel(R) Math Kernel Library Version 2019.0.1 Product Build 20180928 for Intel(R) 64 architecture applications
> * Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
> * OpenMP 201511 (a.k.a. OpenMP 4.5)
> * NNPACK is enabled
> * CPU capability usage: AVX2
> * CUDA Runtime 10.2
> * NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
> * CuDNN 7.6.5
> * Magma 2.5.2
> * Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,
> 
> ## TorchVision: 0.6.0
> OpenCV: 4.1.1
> MMCV: 1.3.5
> MMGen: 0.2.0+
> MMCV Compiler: GCC 5.4
> MMCV CUDA Compiler: 10.2
> 2021-05-31 15:45:07,821 - mmgen - INFO - Distributed training: True
> 2021-05-31 15:45:08,123 - mmgen - INFO - Config:
> model = dict(
> type='Pix2Pix',
> generator=dict(
> type='UnetGenerator',
> in_channels=3,
> out_channels=3,
> num_down=8,
> base_channels=64,
> norm_cfg=dict(type='BN'),
> use_dropout=True,
> init_cfg=dict(type='normal', gain=0.02)),
> discriminator=dict(
> type='PatchDiscriminator',
> in_channels=6,
> base_channels=64,
> num_conv=3,
> norm_cfg=dict(type='BN'),
> init_cfg=dict(type='normal', gain=0.02)),
> gan_loss=dict(
> type='GANLoss',
> gan_type='vanilla',
> real_label_val=1.0,
> fake_label_val=0.0,
> loss_weight=1.0),
> pixel_loss=dict(type='L1Loss', loss_weight=100.0, reduction='mean'))
> train_cfg = dict(direction='b2a')
> test_cfg = dict(direction='b2a', show_input=True)
> train_dataset_type = 'PairedImageDataset'
> val_dataset_type = 'PairedImageDataset'
> img_norm_cfg = dict(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])
> train_pipeline = [
> dict(
> type='LoadPairedImageFromFile',
> io_backend='disk',
> key='pair',
> flag='color'),
> dict(
> type='Resize',
> keys=['img_a', 'img_b'],
> scale=(286, 286),
> interpolation='bicubic'),
> dict(type='FixedCrop', keys=['img_a', 'img_b'], crop_size=(256, 256)),
> dict(type='Flip', keys=['img_a', 'img_b'], direction='horizontal'),
> dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
> dict(
> type='Normalize',
> keys=['img_a', 'img_b'],
> to_rgb=False,
> mean=[0.5, 0.5, 0.5],
> std=[0.5, 0.5, 0.5]),
> dict(type='ImageToTensor', keys=['img_a', 'img_b']),
> dict(
> type='Collect',
> keys=['img_a', 'img_b'],
> meta_keys=['img_a_path', 'img_b_path'])
> ]
> test_pipeline = [
> dict(
> type='LoadPairedImageFromFile',
> io_backend='disk',
> key='pair',
> flag='color'),
> dict(
> type='Resize',
> keys=['img_a', 'img_b'],
> scale=(256, 256),
> interpolation='bicubic'),
> dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
> dict(
> type='Normalize',
> keys=['img_a', 'img_b'],
> to_rgb=False,
> mean=[0.5, 0.5, 0.5],
> std=[0.5, 0.5, 0.5]),
> dict(type='ImageToTensor', keys=['img_a', 'img_b']),
> dict(
> type='Collect',
> keys=['img_a', 'img_b'],
> meta_keys=['img_a_path', 'img_b_path'])
> ]
> data = dict(
> samples_per_gpu=1,
> workers_per_gpu=4,
> drop_last=True,
> train=dict(
> type='PairedImageDataset',
> dataroot='data/paired/facades',
> pipeline=[
> dict(
> type='LoadPairedImageFromFile',
> io_backend='disk',
> key='pair',
> flag='color'),
> dict(
> type='Resize',
> keys=['img_a', 'img_b'],
> scale=(286, 286),
> interpolation='bicubic'),
> dict(
> type='FixedCrop',
> keys=['img_a', 'img_b'],
> crop_size=(256, 256)),
> dict(type='Flip', keys=['img_a', 'img_b'], direction='horizontal'),
> dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
> dict(
> type='Normalize',
> keys=['img_a', 'img_b'],
> to_rgb=False,
> mean=[0.5, 0.5, 0.5],
> std=[0.5, 0.5, 0.5]),
> dict(type='ImageToTensor', keys=['img_a', 'img_b']),
> dict(
> type='Collect',
> keys=['img_a', 'img_b'],
> meta_keys=['img_a_path', 'img_b_path'])
> ],
> test_mode=False),
> val=dict(
> type='PairedImageDataset',
> dataroot='data/paired/facades',
> pipeline=[
> dict(
> type='LoadPairedImageFromFile',
> io_backend='disk',
> key='pair',
> flag='color'),
> dict(
> type='Resize',
> keys=['img_a', 'img_b'],
> scale=(256, 256),
> interpolation='bicubic'),
> dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
> dict(
> type='Normalize',
> keys=['img_a', 'img_b'],
> to_rgb=False,
> mean=[0.5, 0.5, 0.5],
> std=[0.5, 0.5, 0.5]),
> dict(type='ImageToTensor', keys=['img_a', 'img_b']),
> dict(
> type='Collect',
> keys=['img_a', 'img_b'],
> meta_keys=['img_a_path', 'img_b_path'])
> ],
> test_mode=True),
> test=dict(
> type='PairedImageDataset',
> dataroot='data/paired/facades',
> pipeline=[
> dict(
> type='LoadPairedImageFromFile',
> io_backend='disk',
> key='pair',
> flag='color'),
> dict(
> type='Resize',
> keys=['img_a', 'img_b'],
> scale=(256, 256),
> interpolation='bicubic'),
> dict(type='RescaleToZeroOne', keys=['img_a', 'img_b']),
> dict(
> type='Normalize',
> keys=['img_a', 'img_b'],
> to_rgb=False,
> mean=[0.5, 0.5, 0.5],
> std=[0.5, 0.5, 0.5]),
> dict(type='ImageToTensor', keys=['img_a', 'img_b']),
> dict(
> type='Collect',
> keys=['img_a', 'img_b'],
> meta_keys=['img_a_path', 'img_b_path'])
> ],
> test_mode=True))
> checkpoint_config = dict(interval=4000, by_epoch=False, save_optimizer=True)
> log_config = dict(
> interval=100, hooks=[dict(type='TextLoggerHook', by_epoch=False)])
> custom_hooks = [
> dict(
> type='VisualizationHook',
> output_dir='training_samples',
> res_name_list=['fake_b'],
> interval=100)
> ]
> runner = dict(
> type='DynamicIterBasedRunner',
> is_dynamic_ddp=True,
> pass_training_status=True)
> dist_params = dict(backend='nccl')
> log_level = 'INFO'
> load_from = None
> resume_from = None
> workflow = [('train', 1)]
> find_unused_parameters = True
> cudnn_benchmark = True
> dataroot = 'data/paired/facades'
> optimizer = dict(
> generator=dict(type='Adam', lr=0.0002, betas=(0.5, 0.999)),
> discriminator=dict(type='Adam', lr=0.0002, betas=(0.5, 0.999)))
> lr_config = None
> visual_config = None
> total_iters = 80000
> exp_name = 'pix2pix_facades'
> work_dir = './work_dirs/experiments/pix2pix_facades'
> metrics = dict(
> FID=dict(type='FID', num_images=106, image_shape=(3, 256, 256)),
> IS=dict(type='IS', num_images=106, image_shape=(3, 256, 256)))
> gpu_ids = range(0, 1)
> 
> 2021-05-31 15:45:08,124 - mmgen - INFO - Set random seed to 2021, deterministic: False
> fatal: Not a git repository (or any parent up to mount point /home)
> Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
> 2021-05-31 15:45:13,765 - mmgen - INFO - Start running, host: lthpc@lthpc, work_dir: /home/lthpc/mmgeneration-master/work_dirs/experiments/pix2pix_facades
> 2021-05-31 15:45:13,765 - mmgen - INFO - workflow: [('train', 1)], max: 80000 iters
> Traceback (most recent call last):
> File "tools/train.py", line 163, in
> main()
> File "tools/train.py", line 159, in main
> meta=meta)
> File "/home/lthpc/mmgeneration-master/mmgen/apis/train.py", line 196, in train_model
> runner.run(data_loaders, cfg.workflow, cfg.total_iters)
> File "/home/lthpc/mmgeneration-master/mmgen/core/runners/dynamic_iterbased_runner.py", line 284, in run
> iter_runner(iter_loaders[i], **kwargs)
> File "/home/lthpc/mmgeneration-master/mmgen/core/runners/dynamic_iterbased_runner.py", line 214, in train
> outputs = self.model.train_step(data_batch, self.optimizer, **kwargs)
> File "/home/lthpc/anaconda3/lib/python3.7/site-packages/mmcv/parallel/distributed.py", line 51, in train_step
> output = self.module.train_step(*inputs[0], **kwargs[0])
> TypeError: train_step() got an unexpected keyword argument 'running_status'
> Traceback (most recent call last):
> File "/home/lthpc/anaconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
> "**main**", mod_spec)
> File "/home/lthpc/anaconda3/lib/python3.7/runpy.py", line 85, in _run_code
> exec(code, run_globals)
> File "/home/lthpc/anaconda3/lib/python3.7/site-packages/torch/distributed/launch.py", line 263, in
> main()
> File "/home/lthpc/anaconda3/lib/python3.7/site-packages/torch/distributed/launch.py", line 259, in main
> cmd=cmd)
> subprocess.CalledProcessError: Command '['/home/lthpc/anaconda3/bin/python', '-u', 'tools/train.py', '--local_rank=0', 'configs/pix2pix/pix2pix_vanilla_unet_bn_1x1_80k_facades.py', '--launcher', 'pytorch']' returned non-zero exit status 1.
> 
> ```
> A placeholder for trackback.
> ```
> 
> **Bug fix**
> If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

Sorry for the inconvenience. This bug has been fixed in #38.

**Comment by Lasia205565** [comment]
Thank you. It works for me 👍
